esphome/components/sen5x/sensor.py

```diff
@@ -0,0 +1,241 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor, sensirion_common
+from esphome import automation
+from esphome.automation import maybe_simple_id
+
+from esphome.const import (
+    CONF_HUMIDITY,
+    CONF_ID,
+    CONF_OFFSET,
+    CONF_PM_1_0,
+    CONF_PM_10_0,
+    CONF_PM_2_5,
+    CONF_PM_4_0,
+    CONF_STORE_BASELINE,
+    CONF_TEMPERATURE,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_NITROUS_OXIDE,
+    DEVICE_CLASS_PM1,
+    DEVICE_CLASS_PM10,
+    DEVICE_CLASS_PM25,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_VOLATILE_ORGANIC_COMPOUNDS,
+    ICON_CHEMICAL_WEAPON,
+    ICON_RADIATOR,
+    ICON_THERMOMETER,
+    ICON_WATER_PERCENT,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_CELSIUS,
+    UNIT_MICROGRAMS_PER_CUBIC_METER,
+    UNIT_PERCENT,
+)
+
+CODEOWNERS = ["@martgras"]
+DEPENDENCIES = ["i2c"]
+AUTO_LOAD = ["sensirion_common"]
+
+sen5x_ns = cg.esphome_ns.namespace("sen5x")
+SEN5XComponent = sen5x_ns.class_(
+    "SEN5XComponent", cg.PollingComponent, sensirion_common.SensirionI2CDevice
+)
+SEN5XAccelerationMode = sen5x_ns.enum("RHT_Acceleration_Mode")
```

Comment:
This should match the c++ name on both sides
```suggestion
RhtAccelerationMode = sen5x_ns.enum("RhtAccelerationMode")
```